### PR TITLE
Add `ClearMLLogger.v2`

### PR DESCRIPTION
**Pull request by shadeMe**
## Description

This PR refactors commonly reused code into separate functions and adds `ClearMLLogger.v2`. 

`v2` changes/adds the following:
- The logger no longer calls the console logger internally. To avail oneself of both, the `ChainLogger` needs to be used as the main logger.
- `log_custom_stats` has been added as a new optional parameter, which is a list of regexps. Keys in the `info` dictionary that match any of these regexps are automatically logged to ClearML.

### Types of change

Refactor/New feature

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy_loggers/clearml.py [hunk 1]
```diff
@@ -1,9 +1,60 @@
...
-from spacy import Language, util
-from spacy.training.loggers import console_logger
+from spacy import Language
+from spacy.util import SimpleFrozenList
+from .util import dict_to_dot, dot_to_dict, matcher_for_regex_patterns
+from .util import setup_default_console_logger
+
+
+# entry point: spacy.ClearMLLogger.v2
+def clearml_logger_v2(
```
Comment (rmitsch) [inline spacy_loggers/clearml.py:13]: Being a stickler again - can you add docstrings for public functions?

File: README.md [hunk 2]
```diff
@@ -180,6 +180,10 @@ In addition to the above, the following artifacts can also be optionally capture
 - Dataset used to train.
   - Versioned using ClearML Data and linked to under Configuration -> User Properties on the web UI.
 
+`spacy.ClearMLLogger.v1` and below automatically call the default console logger.
+However, starting with `spacy.ClearMLLogger.v2`, console logging must be activated
+through the use of the [ChainLogger](#chainlogger).
+
```
Comment (rmitsch) [inline README.md:186]: Maybe a few words on why this is necessary?

File: spacy_loggers/clearml.py [hunk 3]
```diff
@@ -25,113 +119,150 @@ def clearml_logger_v1(
...
-            )
 
-        def log_step(info: Optional[Dict[str, Any]]):
-            console_log_step(info)
-            if info is not None:
-                score = info.get("score")
-                other_scores = info.get("other_scores")
-                losses = info.get("losses")
-                if score:
+def _log_step_cleaml(
```
Comment (rmitsch) [inline spacy_loggers/clearml.py:179]:
```suggestion
def _log_step_clearml(
```

File: spacy_loggers/clearml.py [hunk 4]
```diff
@@ -25,113 +119,150 @@ def clearml_logger_v1(
...
+                    weights_path=log_best_dir,
+                    auto_delete_file=False,
+                    target_filename="best_model",
+                )
 
-    return setup_logger
+
+def _finalize_clearml(
+    task: Any,
+):
```
Comment (rmitsch) [inline spacy_loggers/clearml.py:252]: Is this how `black` formats the signature?

Comment (shadeMe) [reply spacy_loggers/clearml.py]: Interestingly, `black` kept the above formatting until I changed it be an one-liner.

Comment (rmitsch) [reply spacy_loggers/clearml.py]: Yeah, it's a bit weird sometimes.

**Review by rmitsch — APPROVED** [review]

**Pull request merged**